libr/bin/format/elf/elf.c

```diff
@@ -2872,16 +2898,24 @@ static RBinElfSection *get_sections_from_phdr(ELFOBJ *bin) {
 		num_sections++;
 	}
 
-	ret = calloc (num_sections + 1, sizeof(RBinElfSection));
+	ret = calloc (num_sections + 1, sizeof (RBinElfSection));
 	if (!ret) {
 		return NULL;
 	}
 
-	i = 0;
-	create_section_from_phdr (bin, ret, &i, ".rel.dyn", reldyn, reldynsz);
-	create_section_from_phdr (bin, ret, &i, ".rela.plt", relava, pltgotsz);
-	create_section_from_phdr (bin, ret, &i, ".rel.plt", relva, relasz);
-	create_section_from_phdr (bin, ret, &i, ".got.plt", pltgotva, pltgotsz);
+	size_t i = 0;
+	if (i < num_sections) {
```

Comment:
this series of `if` doesn't look very good... I'm tempted to pass `num_sections` to `create_section_from_phdr` and let that function check if `i < num_sections`.

Reply:
yeah that could be another option... or just use rvector api which already checks for that and remove the original calloc/.last approach

Reply:
actually it should never overflow if the num_sections count is done properly. but there's nohting that checks for integer overflows when couting sections above..